react/LocationSelect.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import { useAddressContext } from 'vtex.address-context/AddressContext'
+import rules from './countries/rules'
+import { Dropdown } from 'vtex.styleguide'
+import { FormattedMessage, defineMessages } from 'react-intl'
+import { Address } from 'vtex.checkout-graphql'
+
+const messages = defineMessages({
+  province: {
+    defaultMessage: '',
+    id: 'place-components.label.province',
+  },
+  city: {
+    defaultMessage: '',
+    id: 'place-components.label.city',
+  },
+  department: {
+    defaultMessage: '',
+    id: 'place-components.label.department',
+  },
+  region: {
+    defaultMessage: '',
+    id: 'place-components.label.region',
+  },
+  community: {
+    defaultMessage: '',
+    id: 'place-components.label.community',
+  },
+  municipality: {
+    defaultMessage: '',
+    id: 'place-components.label.municipality',
+  },
+  district: {
+    defaultMessage: '',
+    id: 'place-components.label.district',
+  },
+  state: {
+    defaultMessage: '',
+    id: 'place-components.label.state',
+  },
+  locality: {
+    defaultMessage: '',
+    id: 'place-components.label.locality',
+  },
+})
+
+const LocationSelect: StorefrontFunctionComponent<{}> = () => {
```

Comment:
if you don't use the `schema` property, I'd rather have this declared as a `React.FC`

Reply:
I addressed all others nits but this one I would rather address in the latest code available for this project, since this is applicable for most of my components 👍